OpenMarket/OpenMarket/Application/Presentation/View/GridCollectionCell.swift

```diff
@@ -0,0 +1,147 @@
+    }()
+    
+    private let leftoverLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.preferredFont(forTextStyle: .caption1)
+        label.textColor = .gray
+        
+        return label
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        configureGridCell()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("not implemnted")
+    }
+    
+    override func prepareForReuse() {
+        super.prepareForReuse()
+        
+        productImageView.image = nil
+        productNameLabel.text = nil
+        originalPriceLabel.text = nil
+        originalPriceLabel.textColor = .systemGray
+        bargainPriceLabel.text = nil
+        bargainPriceLabel.textColor = .systemGray
+        leftoverLabel.text = nil
+        leftoverLabel.textColor = .systemGray
+    }
+    
+}
+
+extension GridCollectionCell {
+    private func configureGridCell() {
+        contentView.addSubview(rootStackView)
+        rootStackView.addArrangedSubview(productImageView)
+        rootStackView.addArrangedSubview(labelStackView)
+        
+        labelStackView.addArrangedSubview(productNameLabel)
+        labelStackView.addArrangedSubview(originalPriceLabel)
+        labelStackView.addArrangedSubview(bargainPriceLabel)
+        labelStackView.addArrangedSubview(leftoverLabel)
+
+        let inset = CGFloat(10)
+            
+        NSLayoutConstraint.activate([
+            rootStackView.leadingAnchor.constraint(equalTo: contentView.leadingAnchor, constant: inset),
+            rootStackView.trailingAnchor.constraint(equalTo: contentView.trailingAnchor, constant: -inset),
+            rootStackView.topAnchor.constraint(equalTo: contentView.topAnchor, constant: inset),
+            rootStackView.bottomAnchor.constraint(equalTo: contentView.bottomAnchor, constant: -inset)
+        ])
+    }
+    
+    func updateUI(_ data: ProductEntity) {
+        productImageView.image = data.thumbnailImage
+        productNameLabel.text = data.name
+        originalPriceLabel.text = data.currency + " " + data.originalPrice.numberFormatter()
+        originalPriceLabel.numberOfLines = 0
+        bargainPriceLabel.text = data.currency + " " + data.discountedPrice.numberFormatter()
+        bargainPriceLabel.numberOfLines = 0
+        leftoverLabel.text = "잔여수량 : " + String(data.stock)
+        
+        if data.originalPrice == data.discountedPrice {
+            bargainPriceLabel.isHidden = true
+            originalPriceLabel.attributedText = originalPriceLabel.text?.strikeThrough(value: 0)
+            originalPriceLabel.textColor = .systemGray
+        } else {
+            bargainPriceLabel.isHidden = false
+            originalPriceLabel.attributedText = originalPriceLabel.text?.strikeThrough(value: NSUnderlineStyle.single.rawValue)
+            originalPriceLabel.textColor = .systemRed
+        }
```

Comment:
```suggestion
        iviewModel.isDiscountedItem == true ?  self.configureForBargain() : self.configureForOriginal()
```
이런식으로 로직을 분리하는게 유지보수에 좋습니다. 보기에도 좋고요. (예를 든거지 참고만 하십시오 ㅎㅎ)
뷰모델이나 모델에 메소드를 추가해보려고 노력해보세요. 
뷰는 최대한 멍청해야 하고 뷰모델, 모델이 로직을 담당해야 하는 말이 있습니다. 

> 뷰는 최대한 멍청해야 한다는 말 
뷰는 화면에 보이게 하는 로직, 유저와 인터렉션이 있을때 뷰컨에 알려야 하는 로직 등 뷰로서의 역할만 하는 코드가 있어야 하고 비즈니스 로직은 뷰모델쪽에 위치해야 합니다. 그래야 유지보수 하기 좋아요.